test/utils.jl

```diff
@@ -74,17 +75,23 @@ function process_eval(expr::Expr; stdout=devnull)
     cmd = `$(Base.julia_cmd()) --project=$(Ray.project_dir()) -e $code`
 
     input = Pipe()
-    err = Pipe()
+    stderr = stderr === devnull ? Pipe() : stderr
     result_file = tempname()
-    p = run(pipeline(cmd; stdin=input, stdout=stdout, stderr=err); wait=false)
+    p = run(pipeline(cmd; stdin=input, stdout, stderr); wait=false)
 
     serialize(input, expr)
     serialize(input, result_file)
     wait(p)
     if success(p)
         return deserialize(result_file)
-    else
-        err_str = String(readavailable(err))
+    elseif stderr isa Pipe || stderr isa IOBuffer
+        err_str = if stderr isa Pipe
+            String(readavailable(stderr))
+        else
+            String(take!(stderr))
```

Comment:
In the event that a caller would pass in `stderr` and wrap `process_eval` in a `try`/`catch` and try to read the passed in `stderr` this would consume the bytes before they have the chance to read them. However, in our specific use case displaying the error message is more important so I went ahead with this.

Reply:
I added a comment about this